server/src/daemon/service.rs

```diff
@@ -0,0 +1,311 @@
+        let mut secrets = HashMap::with_capacity(paths.len());
+        for collection in self.collections.read().await.iter() {
+            let items = collection.items.read().await;
+            for item in items.iter() {
+                for path in paths.iter() {
+                    if item.path() == path.as_ref() {
+                        let secret = item.secret(session.clone()).await?;
+                        secrets.insert(path.to_owned(), secret);
+                    }
+                }
+            }
+        }
+        Ok(secrets)
+    }
+
+    pub async fn read_alias(&self, name: &str) -> ObjectPath {
+        self.collections
+            .read()
+            .await
+            .iter()
+            .find_map(|c| {
+                if c.label() == name {
+                    Some(c.path().to_owned())
+                } else {
+                    None
+                }
+            })
+            .unwrap_or_default()
+    }
+
+    pub async fn set_alias(
+        &self,
+        #[zbus(signal_context)] ctxt: SignalContext<'_>,
+        alias: &str,
+        path: ObjectPath<'_>,
+    ) -> Result<()> {
+        match self
+            .collections
+            .read()
+            .await
+            .iter()
+            .find(|c| c.path() == path)
+        {
+            Some(collection) => {
+                collection.set_alias(&ctxt, alias).await?;
+                Ok(())
+            }
+            None => {
+                tracing::error!("Collection {path} not found");
+                Err(ServiceError::NoSuchObject)
+            }
+        }
+    }
+
+    #[zbus(property, name = "Collections")]
+    pub async fn collections(&self) -> Vec<ObjectPath> {
+        self.collections
+            .read()
+            .await
+            .iter()
+            .map(|collection| collection.path().to_owned())
+            .collect()
+    }
+
+    #[zbus(signal)]
+    pub async fn collection_created(
+        ctxt: &SignalContext<'_>,
+        collection: ObjectPath<'_>,
+    ) -> zbus::Result<()>;
+
+    #[zbus(signal)]
+    pub async fn collection_deleted(
+        ctxt: &SignalContext<'_>,
+        collection: ObjectPath<'_>,
+    ) -> zbus::Result<()>;
+
+    #[zbus(signal)]
+    pub async fn collection_changed(
+        ctxt: &SignalContext<'_>,
+        collection: ObjectPath<'_>,
+    ) -> zbus::Result<()>;
+}
+
+impl Service {
+    pub async fn new() -> Self {
+        Self {
+            collections: RwLock::new(Vec::new()),
+            keyring: Arc::new(Keyring::load_default().await.unwrap()),
+            cnx: Default::default(),
+            manager: Arc::new(Mutex::new(ServiceManager::new())),
+            sessions_counter: RwLock::new(0),
+        }
+    }
+
+    pub async fn run(self) -> Result<()> {
+        let cnx = zbus::connection::Builder::session()?
+            .name(oo7::dbus::api::Service::DESTINATION.unwrap())?
+            //.name("org.freedesktop.secrets.Devel")?
```

Comment:
Instead, define all those values in main.rs as constants, and make them change based on whether you are doing a release or debug build I would say. It would make testing much simpler

Reply:
Implemented via https://github.com/bilelmoussaoui/oo7/pull/73/commits/fe27cd8381955f67709f8bdeb89d16ada70f0f83